src/types.ts

```diff
@@ -37,19 +40,7 @@ export interface RedefinedDomainResolverProps {
   resolverOptions?: ResolverOptions;
   placeholder?: string
 
-  onUpdate(account: Account | null): void;
-}
-
-export interface RedefinedDomainResolverReverseProps {
-  theme?: Theme;
-  width?: string;
-  height?: string;
-  autoFocus?: boolean;
-  disabled?: boolean;
-  resolverOptions?: ResolverOptions;
-  placeholder?: string
-
-  onUpdate(account: Account | null): void;
+  onUpdate(account: Account | ReverseAccount): void;
```

Comment:
Should be wrapped to something like
`type TypedResult = (Account | ReverseAccount) & { type: string }`